Arrowgene.Ddon.GameServer/Handler/JobJobValueShopBuyItemHandler.cs

```diff
@@ -0,0 +1,87 @@
+using Arrowgene.Ddon.GameServer.Characters;
+using Arrowgene.Ddon.Server;
+using Arrowgene.Ddon.Shared.Entity.PacketStructure;
+using Arrowgene.Ddon.Shared.Entity.Structure;
+using Arrowgene.Ddon.Shared.Model;
+using Arrowgene.Ddon.Shared.Network;
+using Arrowgene.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Arrowgene.Ddon.GameServer.Handler
+{
+    internal class JobJobValueShopBuyItemHandler : GameStructurePacketHandler<C2SJobJobValueShopBuyItemReq>
+    {
+        private static readonly ServerLogger Logger = LogProvider.Logger<ServerLogger>(typeof(JobGetPlayPointListHandler));
+
+        private readonly ItemManager _itemManager;
+        private readonly PlayPointManager _playPointManager;
+
+        public JobJobValueShopBuyItemHandler(DdonGameServer server) : base(server)
+        {
+            _itemManager = server.ItemManager;
+            _playPointManager = server.PPManager;
+        }
+
+        public override void Handle(GameClient client, StructurePacket<C2SJobJobValueShopBuyItemReq> packet)
+        {
+            uint boughtAmount = packet.Structure.Num;
+
+            bool sendToItemBag;
+            switch (packet.Structure.StorageType)
+            {
+                case 19:
+                    // If packet.Structure.Destination is 19: Send to corresponding item bag
+                    sendToItemBag = true;
+                    break;
+                case 20:
+                    // If packet.Structure.Destination is 20: Send to storage 
+                    sendToItemBag = false;
+                    break;
+                default:
+                    throw new Exception("Unexpected destination when buying goods: " + packet.Structure.StorageType);
```

Comment:
Throw a Response error exception instead and try grinding the most appropriate ErrorCode for it